contract/src/airdrop.contract.js

```diff
@@ -23,6 +23,18 @@ import { objectToMap } from './helpers/objectTools.js';
 import { getMerkleRootFromMerkleProof } from './merkle-tree/index.js';
 import '@agoric/zoe/exported.js';
 
+const ProofDataShape = harden({
+  hash: M.string(),
+  direction: M.string(),
+});
+
+const OfferArgsShape = harden({
+  tier: M.number(),
+  address: M.string(),
+  key: M.string(),
+  proof: M.arrayOf(ProofDataShape),
+});
```

Comment:
This looks good. Maybe we can also specify max length as (perhaps limits parameter) for `string` and `arrayOf`. WDYT?

Reply:
They have built-in limits.

Reply:
oh... though max length makes sense for `address`, since the app actually knows the lengh.

maybe `proof` has an app-specified max length too; I dunno